.github/actions/lint/action.yml

```diff
@@ -12,7 +12,7 @@ runs:
     - run: brew install clang-format
       shell: bash
       if: runner.os == 'macOS'
-    - run: pip install wheel pre-commit==2.17.0 mypy==0.780 numpy==1.21.6
+    - run: pip install wheel pre-commit==2.17.0 mypy==0.971 numpy==1.21.6 torch==1.11.0 tensorflow==2.7.2
```

Comment:
why do we need to install tensorflow/torch for a linter?

Reply:
Otherwise it will ignore torch.Tensor and tf.Tensor types - it does not add any time to CI pre-commit it looks like.

Reply:
pip install runs for 1 minute instead of 5 seconds with python 3.9 on ubuntu. Can you [cache it with a requirements.txt file](https://github.com/actions/cache/blob/main/examples.md#multiple-oss-in-a-workflow-with-a-matrix) and reference it in the doc?

Reply:
The downloads are cached: https://github.com/microsoft/DeepGNN/runs/7928528362?check_suite_focus=true#:~:text=Collecting%20torch%3D%3D1.11.0,67